File: src/app/types/analysis.ts
// Analysis types — ErrorRecord, InsuranceInsight, AppealTrigger

import type { Severity } from "./line-item";

export type ErrorType =
  | "duplicate"
  | "overpriced"
  | "date_mismatch"
  | "denied_billed"
  | "quantity_anomaly"
  | "unbundled"
  | "implausible"
  | "hidden_fee"
  | "unusual_combination"
  | "other";

export interface ErrorRecord {
  line_item_index: number;
  type: ErrorType;
  message: string;
  severity: Severity;
  suggested_action: string;
  affected_amount?: number;
  reasoning?: string;
}

export type InsightStrength = "strong" | "moderate" | "weak";
export type SuccessLikelihood = "high" | "moderate" | "low";

export interface InsuranceInsight {
  rule: string;
  description: string;
  applicability: string;
  strength: InsightStrength;
  appeal_strategy: string;
}

export interface AppealTrigger {
  trigger: string;
  success_likelihood: SuccessLikelihood;
  reasoning: string;
}

File: backend/models/analysis.py
# Analysis Pydantic Models — Explanation, Error, Insight responses

from pydantic import BaseModel
from typing import List, Optional


class LineItemExplanation(BaseModel):
    line_item_id: str
    explanation: str


class ExplanationResponse(BaseModel):
    overall_summary: str
    line_item_explanations: List[LineItemExplanation]


class ErrorRecord(BaseModel):
    line_item_index: int
    type: str
    message: str
    severity: str
    suggested_action: str
    affected_amount: Optional[float] = None
    reasoning: Optional[str] = None


class ErrorSummary(BaseModel):
    critical: int
    warning: int
    info: int


class ErrorsResponse(BaseModel):
    error_summary: ErrorSummary
    errors: List[ErrorRecord]
    cross_document_issues: List[dict] = []


class InsuranceInsight(BaseModel):
    rule: str
    description: str
    applicability: str
    strength: str  # strong | moderate | weak
    appeal_strategy: str


class AppealTrigger(BaseModel):
    trigger: str
    success_likelihood: str  # high | moderate | low
    reasoning: str


class InsightsResponse(BaseModel):
    insights: List[InsuranceInsight]
    appeal_triggers: List[AppealTrigger]
